dtc: libfdt fails to build with gcc15

### Describe the bug

```
ninja: Entering directory `/home/cschuette/openwrt/snapshot/openwrt/build_dir/target-aarch64_cortex-a72_glibc/dtc-1.7.2/openwrt-build'
[1/41] ccache aarch64-openwrt-linux-gnu-gcc -Ilibfdt/libfdt.so.1.7.2.p -Ilibfdt -I../libfdt -I/home/cschuette/openwrt/snapshot/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-15.2.0_glibc/usr/include -I/home/cschuette/openwrt/snapshot/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-15.2.0_glibc/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -Wpointer-arith -Wcast-qual -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Wshadow -Wsuggest-attribute=format -Wwrite-strings -DFDT_ASSUME_MASK=0 -DNO_YAML -DNO_VALGRIND -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -ffile-prefix-map=/home/cschuette/openwrt/snapshot/openwrt/build_dir/target-aarch64_cortex-a72_glibc/dtc-1.7.2=dtc-1.7.2 -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects -Wl,-z,max-page-size=4096 -Wformat -Werror=format-security -DPIC -fPIC -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -Wl,-z,pack-relative-relocs -fPIC -MD -MQ libfdt/libfdt.so.1.7.2.p/fdt_overlay.c.o -MF libfdt/libfdt.so.1.7.2.p/fdt_overlay.c.o.d -o libfdt/libfdt.so.1.7.2.p/fdt_overlay.c.o -c ../libfdt/fdt_overlay.c
FAILED: [code=1] libfdt/libfdt.so.1.7.2.p/fdt_overlay.c.o
ccache aarch64-openwrt-linux-gnu-gcc -Ilibfdt/libfdt.so.1.7.2.p -Ilibfdt -I../libfdt -I/home/cschuette/openwrt/snapshot/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-15.2.0_glibc/usr/include -I/home/cschuette/openwrt/snapshot/openwrt/staging_dir/toolchain-aarch64_cortex-a72_gcc-15.2.0_glibc/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -Wpointer-arith -Wcast-qual -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Wshadow -Wsuggest-attribute=format -Wwrite-strings -DFDT_ASSUME_MASK=0 -DNO_YAML -DNO_VALGRIND -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -ffile-prefix-map=/home/cschuette/openwrt/snapshot/openwrt/build_dir/target-aarch64_cortex-a72_glibc/dtc-1.7.2=dtc-1.7.2 -ffunction-sections -fdata-sections -flto=auto -fno-fat-lto-objects -Wl,-z,max-page-size=4096 -Wformat -Werror=format-security -DPIC -fPIC -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -Wl,-z,pack-relative-relocs -fPIC -MD -MQ libfdt/libfdt.so.1.7.2.p/fdt_overlay.c.o -MF libfdt/libfdt.so.1.7.2.p/fdt_overlay.c.o.d -o libfdt/libfdt.so.1.7.2.p/fdt_overlay.c.o -c ../libfdt/fdt_overlay.c
../libfdt/fdt_overlay.c: In function 'overlay_fixup_phandle':
../libfdt/fdt_overlay.c:424:21: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  424 |                 sep = memchr(fixup_str, ':', fixup_len);
      |                     ^
../libfdt/fdt_overlay.c:434:21: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
  434 |                 sep = memchr(name, ':', fixup_len);
      |                     ^
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

### OpenWrt version

r34692-f37476a856

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

bcm27xx/bcm2711

### Device

Raspberry Pi 4 Model B Rev 1.5

### Image kind

Self-built image

### Steps to reproduce

_No response_

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

_No response_

### Diffconfig

```text

```

### Terms

- [x] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/package/utils/dtc/Makefile
+++ b/package/utils/dtc/Makefile
@@ -5,11 +5,11 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=dtc
-PKG_VERSION:=1.7.2
+PKG_VERSION:=1.8.1
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_HASH:=f200e5ebd7afd20d4b3804a3085af0870fcf3c194f8d7f0f6985cf8bbb4ac0f4
+PKG_HASH:=454e5cce748ce957c91e7f45e66da2cff5100ddb52155d7a909c43cbeaf1bcb0
 PKG_SOURCE_URL:=@KERNEL/software/utils/dtc
 
 PKG_MAINTAINER:=Yousong Zhou <yszhou4tech@gmail.com>
@@ -93,7 +93,7 @@ MESON_ARGS += \
 	-Dyaml=disabled \
 	-Dvalgrind=disabled \
 	-Dpython=disabled \
-	-Dstatic-build=$(if $(CONFIG_DTC_STATIC_BUILD),true,false)
+	$(if $(CONFIG_DTC_STATIC_BUILD),--prefer-static)
 
 define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/lib

